src/log-common.js

```diff
@@ -9,15 +9,133 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
+const path = require('path');
+const fs = require('fs-extra');
+const chalk = require('chalk');
+const {
+  SimpleInterface,
+  rootLogger,
+  serializeMessage,
+  messageFormatJsonString,
+  messageFormatTechnical,
+  MultiLogger,
+  FileLogger,
+  ConsoleLogger,
+} = require('@adobe/helix-log');
 
-const { Logger } = require('@adobe/helix-shared');
+const colors = {
+  info: 'green',
+  warn: 'yellow',
+  error: 'red',
+};
+/**
+ * Log message filter that removes log entries produced during a progress bar
+ * (fields.progress === true), but only on a tty.
+ */
+const suppressProgress = (fields) => {
+  // eslint-disable-next-line no-underscore-dangle,no-console
+  if (fields.progress && console._stderr.isTTY) {
+    return undefined;
+  }
+  // eslint-disable-next-line no-param-reassign
+  delete fields.progress;
+  return fields;
+};
+/**
+ * remove progress field
+ */
+const filterProgress = (fields) => {
```

Comment:
> could you just explain the reasoning behind having the complex machinery with makeLogger and categories and all?

first of all: this only applies to the cli; not the actions (currently).

the main reason for categories, or logger names, or whatever you call it, is that different users of the logging framework can be identifier. for example the git server `[git]` or the simulator `[hlx]` or the info messages from the cli: `[cli]`.

so to unify this, we have an overall logger concepts, using those categories. also, the message format is different, based on the category. for example the `cli` doesn't log the `info` marker on the console. then, each non-cli category uses its own log file.

we also have a special case for the progress bar, where we tag the messages with `progress: true`. so that they are not logged to the console, but still appear on a non-tty and logfile.

Reply:
> Are we sure this needs to be it's own named function? I think for such a short piece of code, a closure would be fine; this might also give more context as to what this is useful for…

yes, i rather prefer explicit names. also it is used here:

```
        filter: categ === 'cli' ? suppressProgress : filterProgress,
```

which would look ugly with anonymous functions.